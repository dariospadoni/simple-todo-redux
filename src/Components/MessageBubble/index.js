import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import {
  ChatBubble, ChatMessage, ChatText, ChatTimeStamp, ChatAction,
} from '../DesignKit';

const MessageBubble = ({ todo, onDelete, onEdit }) => (
  <ChatBubble>
    <ChatMessage>
      <ChatText>{todo.text}</ChatText>
      <ChatTimeStamp>
        <i className="fa fa-clock" />&nbsp;&nbsp;
        {moment(todo.timeStamp).format('DD/MM/YYYY HH:mm')}
      </ChatTimeStamp>
    </ChatMessage>
    <ChatAction onClick={() => onEdit(todo.id)} title="Edit" className="fa fa-edit" />
    <ChatAction onClick={() => onDelete(todo.id)} title="Delete" className="fa fa-trash-alt" />
  </ChatBubble>);

MessageBubble.propTypes = {
  todo: PropTypes.object,
  onDelete: PropTypes.func,
  onEdit: PropTypes.func,
};

export default MessageBubble;
