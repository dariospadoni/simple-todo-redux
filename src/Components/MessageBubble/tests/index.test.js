import React from 'react';
import { render, mount } from 'enzyme';
import moment from 'moment';
import MessageBubble from '../index';
import {
  ChatMessage, ChatText, ChatTimeStamp, ChatBubble, ChatAction,
} from '../../DesignKit/index';

describe('MessageBubble component', () => {
  const timeStamp = new Date('2018-10-15T12:00:00Z');
  const todo = { id: 'todo-1', timeStamp, text: 'test' };

  it('should render the component', () => {
    const mockFn = jest.fn();
    const component = mount(<MessageBubble todo={todo} onDelete={mockFn} onEdit={mockFn} />);
    expect(component.find(ChatBubble).exists()).toBe(true);
    expect(component.find(ChatMessage).exists()).toBe(true);
    expect(component.find(ChatText).text()).toBe('test');
    expect(component.find(ChatTimeStamp).text().trim()).toBe(moment(timeStamp).format('DD/MM/YYYY HH:mm'));
    expect(component.find(ChatAction).exists()).toBe(true);
    expect(component.find(ChatAction).exists()).toBe(true);
    expect(component.find(ChatAction).length).toBe(2);
    expect(component.find(ChatAction).first().hasClass('fa-edit')).toBe(true);
    expect(component.find(ChatAction).last().hasClass('fa-trash-alt')).toBe(true);
  });

  it('should trigger onEdit when clicking edit button', () => {
    const mockEdit = jest.fn();
    const mockDelete = jest.fn();
    const component = mount(<MessageBubble todo={todo} onDelete={mockDelete} onEdit={mockEdit} />);
    component.find(ChatAction).first().simulate('click');
    expect(mockEdit).toHaveBeenCalled();
    expect(mockDelete).not.toHaveBeenCalled();
  });

  it('should trigger onDelete when clicking delete button', () => {
    const mockEdit = jest.fn();
    const mockDelete = jest.fn();
    const component = mount(<MessageBubble todo={todo} onDelete={mockDelete} onEdit={mockEdit} />);
    component.find(ChatAction).last().simulate('click');
    expect(mockEdit).not.toHaveBeenCalled();
    expect(mockDelete).toHaveBeenCalled();
  });

  it('render snapshot', () => {
    const mockFn = jest.fn();
    const component = render(<MessageBubble todo={todo} onDelete={mockFn} onEdit={mockFn} />);
    expect(component).toMatchSnapshot();
  });
});
