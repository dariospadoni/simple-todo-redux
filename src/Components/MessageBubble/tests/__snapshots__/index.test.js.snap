// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBubble component render snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "DesignKit__ChatBubble-bosrNA kJiYLw",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "DesignKit__ChatMessage-eQcxwN hlucBT",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "DesignKit__ChatText-fRDAjz gMrWZn",
            },
            "children": Array [
              Object {
                "data": "test",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "DesignKit__ChatTimeStamp-hCDHet pBAgx",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "fa fa-clock",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "data": "  15/10/2018 14:00",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "text",
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                Object {
                  "data": "  15/10/2018 14:00",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "fa fa-clock",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "text",
                },
              ],
              "name": "small",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "DesignKit__ChatTimeStamp-hCDHet pBAgx",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "fa fa-clock",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "data": "  15/10/2018 14:00",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "text",
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "data": "  15/10/2018 14:00",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "fa fa-clock",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "text",
              },
            ],
            "name": "small",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "DesignKit__ChatText-fRDAjz gMrWZn",
              },
              "children": Array [
                Object {
                  "data": "test",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "fa fa-edit DesignKit__ChatAction-jXKIhm bOFEhW",
            "title": "Edit",
          },
          "children": Array [],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "fa fa-trash-alt DesignKit__ChatAction-jXKIhm bOFEhW",
              "title": "Delete",
            },
            "children": Array [],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "title": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "title": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "title": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "title": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "fa fa-edit DesignKit__ChatAction-jXKIhm bOFEhW",
          "title": "Edit",
        },
        "children": Array [],
        "name": "a",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "fa fa-trash-alt DesignKit__ChatAction-jXKIhm bOFEhW",
            "title": "Delete",
          },
          "children": Array [],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "title": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "title": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "DesignKit__ChatMessage-eQcxwN hlucBT",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "DesignKit__ChatText-fRDAjz gMrWZn",
              },
              "children": Array [
                Object {
                  "data": "test",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "DesignKit__ChatTimeStamp-hCDHet pBAgx",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "fa fa-clock",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "data": "  15/10/2018 14:00",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "text",
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  Object {
                    "data": "  15/10/2018 14:00",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "class": "fa fa-clock",
                      },
                      "children": Array [],
                      "name": "i",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    "type": "text",
                  },
                ],
                "name": "small",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "DesignKit__ChatTimeStamp-hCDHet pBAgx",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "fa fa-clock",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "data": "  15/10/2018 14:00",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "text",
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                Object {
                  "data": "  15/10/2018 14:00",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "fa fa-clock",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "text",
                },
              ],
              "name": "small",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "DesignKit__ChatText-fRDAjz gMrWZn",
                },
                "children": Array [
                  Object {
                    "data": "test",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "title": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "title": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "fa fa-trash-alt DesignKit__ChatAction-jXKIhm bOFEhW",
          "title": "Delete",
        },
        "children": Array [],
        "name": "a",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "fa fa-edit DesignKit__ChatAction-jXKIhm bOFEhW",
            "title": "Edit",
          },
          "children": Array [],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "DesignKit__ChatMessage-eQcxwN hlucBT",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "DesignKit__ChatText-fRDAjz gMrWZn",
                },
                "children": Array [
                  Object {
                    "data": "test",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "DesignKit__ChatTimeStamp-hCDHet pBAgx",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "fa fa-clock",
                      },
                      "children": Array [],
                      "name": "i",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "data": "  15/10/2018 14:00",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "text",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    Object {
                      "data": "  15/10/2018 14:00",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "class": "fa fa-clock",
                        },
                        "children": Array [],
                        "name": "i",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                      "type": "text",
                    },
                  ],
                  "name": "small",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "DesignKit__ChatTimeStamp-hCDHet pBAgx",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "fa fa-clock",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "data": "  15/10/2018 14:00",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "text",
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  Object {
                    "data": "  15/10/2018 14:00",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "class": "fa fa-clock",
                      },
                      "children": Array [],
                      "name": "i",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    "type": "text",
                  },
                ],
                "name": "small",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "DesignKit__ChatText-fRDAjz gMrWZn",
                  },
                  "children": Array [
                    Object {
                      "data": "test",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "title": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "title": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "title": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "title": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
